daemon/daemon.go

```diff
@@ -141,6 +146,54 @@ func (daemon *Daemon) Features() *map[string]bool {
 	return &daemon.configStore.Features
 }
 
+// NewResolveOptionsFunc returns a call back function to resolve "registry-mirrors" and
+// "insecure-registries" for buildkit
+func (daemon *Daemon) NewResolveOptionsFunc() resolver.ResolveOptionsFunc {
+	return func(ref string) docker.ResolverOptions {
+		var (
+			registryKey = "docker.io"
+			mirrors     = make([]string, len(daemon.configStore.Mirrors))
+			m           = map[string]resolver.RegistryConf{}
+		)
+		// must trim "https://" prefix
+		for i, v := range daemon.configStore.Mirrors {
+			mirrors[i] = strings.Split(v, "//")[1]
+		}
+		// set "registry-mirrors"
+		m[registryKey] = resolver.RegistryConf{Mirrors: mirrors}
+		// set "insecure-registries"
+		for _, v := range daemon.configStore.InsecureRegistries {
+			m[v] = resolver.RegistryConf{
+				PlainHTTP: true,
```

Comment:
Does this need to get set if the mirror prefix is "http://"?

Reply:
@dmcgowan Ah I think I the "http://" prefix must be trimmed here as well. Otherwise, buildkit will add an extra prefix.